### PR TITLE
Add snapshot tests using Verify

### PR DESCRIPTION
**Pull request by keahpeters**
# Add snapshot tests using Verify

## The issue or feature being addressed

Possible solution for https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2811.

## Details on the issue fix or feature implementation

This PR is to add snapshot testing for the generated schemas. This PR only adds tests for some of the functionality but if this approach is accepted we could potentially add snapshot tests for everything that is tested in [SwaggerGeneratorTests](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/master/test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGenerator/SwaggerGeneratorTests.cs).

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -27,8 +27,9 @@
     <PackageVersion Include="Nswag.MSbuild" Version="14.0.7" />
     <PackageVersion Include="ReportGenerator" Version="5.2.5" />
     <PackageVersion Include="System.Text.Json" Version="4.6.0" />
+    <PackageVersion Include="Verify.Xunit" Version="24.2.0" />
```
Comment (martincostello) [inline Directory.Packages.props:30]: Could you add this to the dependabot group for xunit please?

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/ace3015e98631f02c05f66ff294158dcdf2ccaf2/.github/dependabot.yml#L33-L35

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGeneratorVerifyTests/SwaggerGeneratorVerifyTests.cs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.TestSupport;
+using VerifyXunit;
+using Xunit;
+
+namespace Swashbuckle.AspNetCore.SwaggerGen.Test
```
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGeneratorVerifyTests/SwaggerGeneratorVerifyTests.cs:17]: As this is a new file, could you change it to use file-scoped namespaces please?

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGeneratorVerifyTests/SwaggerGeneratorVerifyTests.cs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                    SwaggerDocs = new Dictionary<string, OpenApiInfo>
+                    {
+                        ["v1"] = new OpenApiInfo { Version = "V1", Title = "Test API" }
+                    }
+                }
+            );
+
+            var document = subject.GetSwagger("v1");
+
+            return Verifier.Verify(document);
```
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGeneratorVerifyTests/SwaggerGeneratorVerifyTests.cs:47]: I've only briefly played with Verify before - what's the developer experience like locally and in CI if a test fails, or if a snapshot needs regenerating?

Comment (keahpeters) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGeneratorVerifyTests/SwaggerGeneratorVerifyTests.cs:47]: Locally it provides a positive developer experience. If the received file doesn't match the snapshot then it opens both files in your diff tool automatically when the test fails. If you want to keep the new received file as the snapshot you can just copy and paste it from the received file to the verified file there and re-run the test. You need to do this every time you add a new test as the verified file will be empty. 

I haven't checked on CI yet, I will push a change to cause a failure to this PR to check.

Comment (keahpeters) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGeneratorVerifyTests/SwaggerGeneratorVerifyTests.cs]: Both files are shown if there's a failure on CI:

```
NotEqual:
  - Received: SwaggerGeneratorVerifyTests.ApiDescriptionsWithMatchingGroupName.DotNet8_0.received.txt
    Verified: SwaggerGeneratorVerifyTests.ApiDescriptionsWithMatchingGroupName.verified.txt

FileContent:

NotEqual:

Received: SwaggerGeneratorVerifyTests.ApiDescriptionsWithMatchingGroupName.DotNet8_0.received.txt
{
  Info: {
    Title: Test API,
    Version: V1
  },
  Paths: {
    /resource: {
      Operations: {
        Get: {
          Tags: [
            {
              Name: Fake,
              UnresolvedReference: false
            }
          ],
          Responses: {
            200: {
              Description: OK,
              UnresolvedReference: false
            }
          },
          Deprecated: false
        },
        Post: {
          Tags: [
            {
              Name: Fake,
              UnresolvedReference: false
            }
          ],
          Responses: {
            200: {
              Description: OK,
              UnresolvedReference: false
            }
          },
          Deprecated: false
        }
      },
      UnresolvedReference: false
    }
  },
  Components: {},
  HashCode: 0F1036F23326E2BD7543491DEC0E03ABCA5BC97AF845BEE30C164155D616F306F6F6E21A77507DA6BE2A85DF1A17B1F07AE23E7608E37C971BAF1CBA2491E899
}
Verified: SwaggerGeneratorVerifyTests.ApiDescriptionsWithMatchingGroupName.verified.txt
{
  Info: {
    Title: Test API,
    Version: V1
  },
  Paths: {
    /resource: {
      Operations: {
        Get: {
          Tags: [
            {
              Name: Fake,
              UnresolvedReference: false
            }
          ],
          Responses: {
            200: {
              Description: OK,
              UnresolvedReference: false
            }
          },
          Deprecated: false
        },
        Post: {
          Tags: [
            {
              Name: Fake,
              UnresolvedReference: false
            }
          ],
          Responses: {
            200: {
              Description: OK
            }
          },
          Deprecated: false
        }
      },
      UnresolvedReference: false
    }
  },
  Components: {},
  HashCode: 0F1036F23326E2BD7543491DEC0E03ABCA5BC97AF845BEE30C164155D616F306F6F6E21A77507DA6BE2A85DF1A17B1F07AE23E7608E37C971BAF1CBA2491E899
}
```

File: .github/dependabot.yml [hunk 4]
```diff
@@ -33,6 +33,7 @@ updates:
     xunit:
       patterns:
         - xunit*
+        - Verify.Xunit*
```
Comment (martincostello) [inline .github/dependabot.yml:36]:
```suggestion
        - Verify.Xunit
```

**Review by martincostello — COMMENTED** [review]
This looks great!

**Comment by martincostello** [comment]
@keahpeters Is your plan to do additional pull requests after this gets merged to migrate over things from `SwaggerGeneratorTests`, or something else (if anything)?

**Comment by keahpeters** [comment]
> @keahpeters Is your plan to do additional pull requests after this gets merged to migrate over things from `SwaggerGeneratorTests`, or something else (if anything)?

Yeah I can migrate the rest of the tests in another PR. Also I'm not sure if would be better to have these tests in their own project. I added them in the same project for now as they are using some of the existing test fixtures.

**Comment by martincostello** [comment]
Ok cool.

Let's stick with the same project for now, as I was thinking once everything that seems sensible has been migrated, we'd remove the tests they duplicate. If that's the direction we head in, the need to consider having a separate project will probably go away.

**Review by martincostello — APPROVED** [review]
